src/west/manifest-schema.yml

```diff
@@ -95,6 +95,21 @@ mapping:
           path:
             required: false
             type: str
+          # Submodules to checkout within the project
+          submodules:
+            required: false
+            type: seq
```

Comment:
I think it would be good to have this be a boolean, too. That would let people do something simple, like:

```yaml
- name: myproject
  submodules: true
```

To get all the submodules without having to specify them by hand.

You can do this by setting the schema type to `any` and checking the type at runtime using `isinstance`. You can see the example code for `import:` handling in manifest.py for details.

Reply:
I did, as you suggested and I also added another test to check whether using submodules: true results in updating all project submodules.